### PR TITLE
Remove state from abstract interpreter

### PR DESCRIPTION
**Pull request by fridis**
Remove fields `_fuir`, `_cl`, `_pre` and `_s` from `ProcessExpression` . Had to add `site` parameter to `ProcessExpression.env` and `ProcessExpression.box`. 

Added `@Override` annotations to implementations of `ProcessExpression`. Renamed `codeAtAsPos` as `sitePos`. Removed unused imporant and unused `unbox()` in DFA's `ProcessExpression` implementation.

### CODE REVIEW

File: src/dev/flang/fuir/analysis/AbstractInterpreter.java [hunk 1]
```diff
@@ -257,13 +225,26 @@ public RESULT drop(VALUE v, int type)
...
     /**
      * Create a tagged value of type newcl from an untagged value for type valuecl.
+     *
+     * @param s site of the match
+     *
+     * @param value the value that is to be tagged
+     *
+     * @param newcl the clazz id of the new choice type of the tagged value
+     *
+     * @param tagNum the number of the of the choice types's variant value is
```
Comment (michaellilltokiwa) [inline src/dev/flang/fuir/analysis/AbstractInterpreter.java:235]: of the of the

File: src/dev/flang/fuir/analysis/AbstractInterpreter.java [hunk 2]
```diff
@@ -587,13 +567,14 @@ Pair<VALUE,RESULT> processContract(int cl, FUIR.ContractKind ck)
...
-         _fuir.clazzContract(cl, ck, ci) != -1;
+         _fuir.clazzContract(cl, ck, ci) != NO_SITE;
          ci++)
       {
         var s = _fuir.clazzContract(cl, ck, ci);
         var stack = new Stack<VALUE>();
-        for (; !containsVoid(stack) && _fuir.withinCode(s); s = s + _fuir.codeSizeAt(s))
+        for (var si = s; !containsVoid(stack) && _fuir.withinCode(si); si = si + _fuir.codeSizeAt(si))
           {
+            s = si; // when exiting this loop, s will be the site of the last expression, while si no be within the code
```
Comment (michaellilltokiwa) [inline src/dev/flang/fuir/analysis/AbstractInterpreter.java:577]: while si no be within the code

**Review by michaellilltokiwa — APPROVED** [review]
Cool that AbstractInterpreter is stateless again!

File: src/dev/flang/fuir/analysis/AbstractInterpreter.java (hunk 1 above)
Comment (fridis) [inline src/dev/flang/fuir/analysis/AbstractInterpreter.java:235]:
```suggestion
     * @param tagNum the number of the choice types's variant value is
```

File: src/dev/flang/fuir/analysis/AbstractInterpreter.java (hunk 2 above)
Comment (fridis) [inline src/dev/flang/fuir/analysis/AbstractInterpreter.java:577]:
```suggestion
            s = si; // when exiting this loop, s will be the site of the last expression, while si will not be within the code
```

**Pull request merged**